rllib/utils/replay_buffers/episode_replay_buffer.py

```diff
@@ -0,0 +1,508 @@
+                ejected_idx = self.episode_id_to_index[ejected_eps.id_]
+                del self.episode_id_to_index[ejected_eps.id_]
+
+                # All timestep indices that this episode owned.
+                new_indices = []  # New indices that will replace self._indices.
+                idx_cursor = 0
+                for i, idx_tuple in enumerate(self._indices):
+                    if idx_cursor is not None and idx_tuple[0] == ejected_idx:
+                        new_indices.extend(self._indices[idx_cursor:i])
+                        idx_cursor = None
+                    elif idx_cursor is None:
+                        if idx_tuple[0] != ejected_idx:
+                            idx_cursor = i
+                        # Early-out: We reached the end of the to-be-ejected episode.
+                        # We can stop searching further here.
+                        elif idx_tuple[1] == ejected_eps_len - 1:
+                            assert self._indices[i+1][0] != idx_tuple[0]
+                            idx_cursor = i + 1
+                            break
+                if idx_cursor is not None:
+                    new_indices.extend(self._indices[idx_cursor:])
+                self._indices = new_indices
+
+                # Increase episode ejected counter.
+                self._num_episodes_ejected += 1
+
+    @override(ReplayBufferInterface)
+    def sample(
+        self,
+        num_items: Optional[int] = None,
+        *,
+        batch_size_B: Optional[int] = None,
+        batch_length_T: Optional[int] = None,
+    ) -> SampleBatchType:
+        """Returns a batch of size B (number of "rows"), where each row has length T.
+
+        Each row contains consecutive timesteps from an episode, but might not start
+        at the beginning of that episode. Should an episode end within such a
+        trajectory, a random next episode (starting from its t0) will be concatenated
+        to that "row". For more details, see the docstring of the EpisodeReplayBuffer
+        class.
+
+        Args:
+            num_items: See `batch_size_B`. For compatibility with the
+                `ReplayBufferInterface` abstract base class.
+            batch_size_B: The number of rows (trajectories) to return in the batch.
+            batch_length_T: The length of each row (in timesteps) to return in the
+                batch.
+
+        Returns:
+            The sampled batch (observations, actions, rewards, terminateds, truncateds)
+                of dimensions [B, T, ...].
+        """
+        if num_items is not None:
+            assert batch_size_B is None, (
+                "Cannot call `sample()` with both `num_items` and `batch_size_B` "
+                "provided! Use either one."
+            )
+            batch_size_B = num_items
+        # Use our default values if no sizes/lengths provided.
+        batch_size_B = batch_size_B or self.batch_size_B
+        batch_size_T = batch_size_T or self.batch_size_T
```

Comment:
There is no `batch_size_T`, only a `batch_length_T`.

Reply:
Great catch! Fixed.